### PR TITLE
refactor(anta): Refactor VerifyLLDPNeighbors test for nicer failure message

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactor VerifyLLDPNeighbors test for nicer failure message

Related to #587 #388 

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: tests/units/anta_tests/test_connectivity.py [hunk 1]
```diff
@@ -256,16 +250,17 @@
                 },
             },
         ],
+        "inputs": {
+            "neighbors": [
+                {"port": "Ethernet1", "neighbor_device": "DC1-SPINE1", "neighbor_port": "Ethernet1"},
+                {"port": "Ethernet2", "neighbor_device": "DC1-SPINE2", "neighbor_port": "Ethernet1"},
+            ],
+        },
```
Comment (carl-baillargeon) [inline tests/units/anta_tests/test_connectivity.py:258]: Why were these reordered? Just curious.

Comment (vitthalmagadum) [reply tests/units/anta_tests/test_connectivity.py:258]: Just to be consistent.
We have inputs after the eos_data all the places.

File: anta/tests/connectivity.py [hunk 2]
```diff
@@ -111,49 +113,26 @@ class Input(AntaTest.Input):
         neighbors: list[Neighbor]
         """List of LLDP neighbors."""
 
-        class Neighbor(BaseModel):
```
Comment (carl-baillargeon) [inline anta/tests/connectivity.py:114]: Need to add `ClassVar` for backward compatibility.

File: anta/tests/connectivity.py [hunk 3]
```diff
@@ -111,49 +113,26 @@ class Input(AntaTest.Input):
...
-        else:
-            failure_messages = []
-            for failure_type, ports in failures.items():
-                ports_str = "\n   ".join(ports)
-                failure_messages.append(f"{failure_type}:\n   {ports_str}")
-            self.result.is_failure("\n".join(failure_messages))
+            )
+            if not match_found:
+                failure_msg = [f"{info['systemName']}/{info['neighborInterfaceInfo']['interfaceId_v2']}" for info in lldp_neighbor_info]
+                self.result.is_failure(f"{neighbor} - Wrong LLDP neighbors on the ports; {', '.join(failure_msg)}")
```
Comment (carl-baillargeon) [inline anta/tests/connectivity.py:138]:
```suggestion
                self.result.is_failure(f"{neighbor} - Wrong LLDP neighbors on the port; {', '.join(failure_msg)}")
```

File: anta/input_models/connectivity.py [hunk 4]
```diff
@@ -39,3 +39,25 @@ def __str__(self) -> str:
...
+
+    def __str__(self) -> str:
+        """Return a human-readable string representation of the Neighbor for reporting.
+
+        Examples
+        --------
+        Port Ethernet1 (Neighbor: DC1-SPINE2, Neighbor port: Ethernet2)
+
+        """
+        return f"Port {self.port} (Neighbor: {self.neighbor_device}, Neighbor port: {self.neighbor_port})"
```
Comment (carl-baillargeon) [inline anta/input_models/connectivity.py:63]:
```suggestion
        return f"{self.port} (Neighbor: {self.neighbor_device}, Neighbor port: {self.neighbor_port})"
```

File: anta/input_models/connectivity.py [hunk 5]
```diff
@@ -39,3 +39,25 @@ def __str__(self) -> str:
...
+    """The system name of the LLDP neighbor device."""
+    neighbor_port: Interface
+    """The LLDP port on the neighboring device."""
+
+    def __str__(self) -> str:
+        """Return a human-readable string representation of the Neighbor for reporting.
+
+        Examples
+        --------
+        Port Ethernet1 (Neighbor: DC1-SPINE2, Neighbor port: Ethernet2)
```
Comment (carl-baillargeon) [inline anta/input_models/connectivity.py:60]:
```suggestion
        Ethernet1 (Neighbor: DC1-SPINE2, Neighbor port: Ethernet2)
```

File: anta/tests/connectivity.py [hunk 6]
```diff
@@ -111,49 +113,26 @@ class Input(AntaTest.Input):
...
-
         for neighbor in self.inputs.neighbors:
             if neighbor.port not in output:
-                failures.setdefault("Port(s) not configured", []).append(neighbor.port)
+                self.result.is_failure(f"{neighbor} - Port not found")
                 continue
 
             if len(lldp_neighbor_info := output[neighbor.port]["lldpNeighborInfo"]) == 0:
-                failures.setdefault("No LLDP neighbor(s) on port(s)", []).append(neighbor.port)
+                self.result.is_failure(f"{neighbor} - No LLDP neighbors on the port")
```
Comment (carl-baillargeon) [inline anta/tests/connectivity.py:128]:
```suggestion
                self.result.is_failure(f"{neighbor} - No LLDP neighbors")
```

File: anta/tests/connectivity.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/tests/connectivity.py:114]: Added. Thanks!!

File: anta/tests/connectivity.py (hunk 3 above)
Comment (vitthalmagadum) [reply anta/tests/connectivity.py]: Updated

File: anta/input_models/connectivity.py (hunk 4 above)
Comment (vitthalmagadum) [reply anta/input_models/connectivity.py]: Updated.

File: anta/input_models/connectivity.py (hunk 5 above)
Comment (vitthalmagadum) [reply anta/input_models/connectivity.py]: Updated

File: anta/tests/connectivity.py (hunk 6 above)
Comment (vitthalmagadum) [reply anta/tests/connectivity.py]: Updated

**Review by carl-baillargeon — APPROVED** [review]
LGTM

File: anta/input_models/connectivity.py [hunk 7]
```diff
@@ -39,3 +39,25 @@ def __str__(self) -> str:
         """
         df_status = ", df-bit: enabled" if self.df_bit else ""
         return f"Host {self.destination} (src: {self.source}, vrf: {self.vrf}, size: {self.size}B, repeat: {self.repeat}{df_status})"
+
+
+class Neighbor(BaseModel):
```
Comment (gmuloc) [inline anta/input_models/connectivity.py:44]: please let's call this LLDPNeighbor

File: anta/tests/connectivity.py [hunk 8]
```diff
@@ -110,50 +112,28 @@ class Input(AntaTest.Input):
...
-        else:
-            failure_messages = []
-            for failure_type, ports in failures.items():
-                ports_str = "\n   ".join(ports)
-                failure_messages.append(f"{failure_type}:\n   {ports_str}")
-            self.result.is_failure("\n".join(failure_messages))
+            )
+            if not match_found:
+                failure_msg = [f"{info['systemName']}/{info['neighborInterfaceInfo']['interfaceId_v2']}" for info in lldp_neighbor_info]
+                self.result.is_failure(f"{neighbor} - Wrong LLDP neighbors; {', '.join(failure_msg)}")
```
Comment (gmuloc) [inline anta/tests/connectivity.py:139]:
```suggestion
                self.result.is_failure(f"{neighbor} - Wrong LLDP neighbors: {', '.join(failure_msg)}")
```

File: anta/tests/connectivity.py [hunk 9]
```diff
@@ -77,14 +74,19 @@ def test(self) -> None:
 
 
 class VerifyLLDPNeighbors(AntaTest):
-    """Verifies that the provided LLDP neighbors are present and connected with the correct configuration.
+    """Verifies the connection status of the specified LLDP (Link Layer Discovery Protocol) neighbors.
+
+    This test performs the following checks for each specified LLDP neighbor:
+
+      1. Confirming matching ports on both local and neighboring devices.
```
Comment (gmuloc) [inline anta/tests/connectivity.py:81]: while we are refactoring I would add somewhere that if the given info is. match potentially we dont raise if any extra neighbor is detected on a given interfacer

e.g.

input Ethernet1 connected to Foo on Ethernet2

if connected to Foo on both Ethernet2 and Ethernet3 then this is accepted

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: anta/input_models/connectivity.py (hunk 7 above)
Comment (vitthalmagadum) [reply anta/input_models/connectivity.py]: Updated.
Thanks!!

File: anta/tests/connectivity.py (hunk 8 above)
Comment (vitthalmagadum) [reply anta/tests/connectivity.py]: Updated

File: anta/tests/connectivity.py (hunk 9 above)
Comment (vitthalmagadum) [reply anta/tests/connectivity.py:81]: Added docstring.

**Pull request merged**